app/src/main/java/org/mozilla/tv/firefox/pinnedtile/PinnedTileViewModel.kt

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pinnedtile
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MediatorLiveData
+import android.arch.lifecycle.Transformations
+import android.arch.lifecycle.ViewModel
+
+/**
+ * TODO
+ *
+ * AndroidViewModel() is a context aware ViewModel => Used for PinnedTileRepo
+ */
+class PinnedTileViewModel(private val pinnedTileRepo: PinnedTileRepo) : ViewModel() {
+
+    private val _tilesList = MediatorLiveData<MutableList<PinnedTile>>()
+    val isEmpty: LiveData<Boolean> = Transformations.map(_tilesList) { input -> input.isEmpty() }
+
+    init {
+        _tilesList.addSource(pinnedTileRepo.loadTilesCache()) { pinnedTileMap ->
+            val feedTileList: MutableList<PinnedTile>? = pinnedTileMap?.values?.toMutableList()
+            if ((feedTileList) != null) {
```

Comment:
We check for null in the VM layer: why is this necessary? Can we just expose the tilesList value directly?

Reply:
Technically never. Old Java habit 😞 . Will update